qbft/spectest/generate/main.go

```diff
@@ -92,7 +93,7 @@ func writeJsonStateComparison(name, testType string, post interface{}) {
 
 	file := filepath.Join(scDir, fmt.Sprintf("%s.json", name))
 	log.Printf("writing state comparison json: %s\n", file)
-	if err := os.WriteFile(file, byts, 0644); err != nil {
+	if err := os.WriteFile(file, byts, 0600); err != nil {
```

Comment:
I would update to 0444

The reason is that we don't want anyone writing to the file, to make tests pass/fail.
There's nothing confidential here so might as well let everyone read

Reply:
Yup, I agree

Reply:
Actually, `gosec` still complains about 0444:
> G306 (CWE-276): Expect WriteFile permissions to be 0600 or less (Confidence: HIGH, Severity: MEDIUM)

Can't it be 0400? Why do we need to allow "read" for group and others?

Reply:
hmm 
if CI passes then it can be 0400 honestly
I just think that there is no confidential information and the SSV huts generate jsons themselves.

I just saw no sec concern so you can choose

Reply:
I also don't see security concerns, but `gosec` complains and I also don't see a reason for the group and others read permission.
Updating it to 0400